### ISSUE TITLE
PEP8

### ISSUE THREAD

**Issue by mictab** [armada-ths/ais#41]
Let's try to keep the code PEP8 :-)

**Comment by macaullyjames** [comment]
I like this.

**Comment by macaullyjames** [comment]
@armada-ths/system-developers we _could_ set this up as an integration (like Travis) so that incoming PRs are style checked automatically. I'm not really all that enthusiastic about enforcing a strict style guide (I'd rather we just agree to try and write readable code) but it's a possibility. Opinions? :)

**Comment by purmonen** [comment]
I'm not sure about the requirement of 79 characters. That seems to be a relic from a time when computers had really low resolution. In order to conform to this lots of code has to be changed in a way that in my opinion doesn't make it more readable.

**Comment by purmonen** [comment]
Adding commas always in dictionaries is ok but making it a rule seems a bit awkward. Making the previous developer do more work by adding a comma in order to save the next maybe non-existing developer a microsecond by not needing to add a comma seems like a huge amount of premature optimisation.

**Comment by purmonen** [comment]
If we are going to follow strict code style rules then of course it might be easier to just follow pep8 instead of debating new rules even if the rules of pep8 are bad at times.

**Comment by purmonen** [comment]
Even adding this decorator for requiring certain permissions for a view breaks the 79 character line length rule (it's 88):
@permission_required('recruitment.view_recruitment_applications', raise_exception=True)

**Comment by macaullyjames** [comment]
I don't think we need to strictly enforce the rules, for me it's more a way of saying "let's not write crappy code" than conforming to a specific subset of Python.

**Comment by macaullyjames** [comment]
Let me rephrase that: I think that it's OK to break the rules **for good reason**. If there's really no way to rewrite a long decorator in a more elegant way so that it conforms to PEP-8, fine, ignore the 79 char limit. On other hand, there's no point in saying "let's conform to PEP-8" and ignoring the rules whenever we feel like it.

**Comment by walaro** [comment]
The 78 (or 79 in the case of pep i guess?) char thing is not about resolution, it's about fitting into standard terminal windows, which are 80*25 chars.

Its also useful for terminal based diff's etc.
I'm not saying we have to use 78/79 columns, but there is very good reason to do so.
